Tests/SentryTests/Profiling/SentryProfilerSwiftTests.swift

```diff
@@ -43,6 +62,49 @@ class SentryProfilerSwiftTests: XCTestCase {
 #endif
     }
 
+#if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+    func testUIEventTrackerProfiling() {
+        fixture.options.profilesSampleRate = 1.0
+        fixture.options.tracesSampleRate = 1.0
+        fixture.eventTracker.start()
+        SentrySDK.setCurrentHub(fixture.hub)
+
+        let exp = expectation(description: "send action finishes")
+        fixture.swizzleWrapper.swizzleSendAction({ _, _, _, _ in
+            exp.fulfill()
+        }, forKey: "first")
```

Comment:
`h`: The SDK will discard this transaction as it has no child spans. Why do you want to send a profile for it?

Reply:
Beside the implementation detail about child spans, why wouldn't we want the SDK to record such a thing? Do we simply not want to actually send transactions for UI events like button presses? 

But on the point about it being dropped, one of my code changes I think changes behavior, because the codepath leads all the way to `-[SentryHub captureTransaction:withScope:]`:
![image](https://user-images.githubusercontent.com/3241469/199876395-49671528-a3dd-43df-8b8d-13597c1cbe2e.png)

Reply:
> Do we simply not want to actually send transactions for UI events like button presses?

Only if they transaction doesn't contain any child spans. A transaction of a button click without any child spans is useless. What value does it add to the product? Yeah, the user clicked a button, but nothing happened. This doesn't help you to improve the performance of your app. 
Instead, if the user clicks a button, that triggers HTTP requests, some file I/O or db queries, the SDK attaches spans to this transaction. Then the transaction contains some information.

Reply:
I guess this comes down to the difference between our performance and profiling features. Perf is interested in those specific functionalities you mention like http or file i/o, whereas profiling is also interested in raw CPU. Maybe there’s room to give us both what we want here 🤔

Reply:
OK, removed the need to keep the reference to the transaction. See https://github.com/getsentry/sentry-cocoa/pull/2228/commits/211a5d1925d269749faef441878226b727eb229a